src/sidebar/components/annotation-omega.js

```diff
@@ -62,9 +66,19 @@ function AnnotationOmega({
     createDraft(annotation, { ...draft, text });
   };
 
+  const onSave = () => {
+    // TODO flash errors if not authenticated or if annotation is empty?
+    setIsSaving(true);
+    annotationsService
+      .save(annotation)
+      .catch(err => flash.error(err.message, 'Saving annotation failed'))
```

Comment:
The current implementation currently explicitly "goes back into edit mode" in the `catch`:

https://github.com/hypothesis/client/blob/ec63961a743dc15e7112290816e19b38ac5d73e5/src/sidebar/components/annotation.js#L299

But that shouldn't be necessary as "edit mode" is triggered by having a draft (which will not have been removed as the annotation didn't successfully save) and not being `isSaving` (which is handled by the `finally`).